syftbox/client/plugins/sync/local_state.py

```diff
@@ -34,6 +35,28 @@ class LocalState(BaseModel):
     def for_client(cls: Type[Self], client: SyftClientInterface) -> Self:
         return cls(path=client.workspace.plugins / LOCAL_STATE_FILENAME)
 
+    def insert_completed_action(self, action: SyncAction) -> None:
+        """Insert action result into local state."""
+        if action.action_type == SyncActionType.NOOP:
+            return
+
+        if action.status == SyncStatus.PROCESSING:
+            raise ValueError("Attempted to insert an action into LocalState that is still being processed.")
+
+        if action.status == SyncStatus.SYNCED:
+            self.insert_synced_file(
```

Comment:
just out of interest, what is the difference here?

Reply:
Just moved this to a different place, and added a check so we dont insert PROCESSING accidentally